hls4ml/templates/vivado_template.py

```diff
@@ -465,6 +476,30 @@ def set_closest_reuse_factor(self, layer):
             print('WARNING: Invalid ReuseFactor={} with "Resource" strategy in layer "{}". Using ReuseFactor={} instead. Valid ReuseFactor(s): {}.'
                 .format(chosen_rf, layer.name, closest_rf, ','.join(map(str, valid_rf))))
             layer.reuse_factor = closest_rf
+    
+    def set_target_reuse_factor(self, layer):
+        targ_cycles = layer.target_cycles
+
+        shuffle_cycles = 6 # Number of clock cycles to move data around
+        if targ_cycles is not None:
+            if layer.get_attr('class_name') == 'Dense': 
+                kernel_multiplies = layer.get_attr('n_out')
+            elif layer.get_attr('class_name') == 'Conv1D':  
+                kernel_multiplies = layer.get_attr('out_width')
+            elif layer.get_attr('class_name') == 'Conv2D': 
+                kernel_multiplies = layer.get_attr('out_height') * layer.get_attr('out_width')
+            else: 
+                print('Target clock unsupported layer')
+                return
```

Comment:
QKeras layers are not supported by this. They have `'class_name'` attributes like `QDense`, `QConv2D`. Is it deliberate that they are not supported? It seems like it should apply to those too. 
If they should be supported, the condition would become either
`if layer.__class__.__name__ == 'Dense'` or `if 'Dense' in layer.get_attr('class_name')`.
It should support the `QConv2DBatchNorm` layer too, if it does apply to QKeras layers.

Reply:
If QKeras layers end up converting to the Dense, Conv1D, or Conv2D layer implementations, then the target cycles parameter should work for them. I'll make some edits to support the QKeras layers.

Reply:
`class_name` attribute comes from the converter and corresponds to the Keras layer class name, it may not match the hls4ml class name. A safer check would be e.g., `'Dense' in layer.__class__.__name__'`